src/api/endpoints/auth/TokenManager.ts

```diff
@@ -32,19 +32,20 @@ export default class TokenManager extends ApiEndpoint {
 
   public async getAccessToken(source: string = "unknown"): Promise<string | undefined> {
     this.log?.debug(`getAccessToken: New token requested from ${source}`);
-    if (this.accessToken !== undefined && this.accessTokenExpiration !== undefined) {
+    if ((this.accessToken !== undefined || this.config.cookieAuth) && this.accessTokenExpiration !== undefined) {
       const exp = Math.trunc(new Date(this.accessTokenExpiration).getTime() / 1000);
       const now = Math.trunc(Date.now() / 1000);
 
       // check expiration minus 5 minutes to guard against race condition or timing issues creating unnecessary 403s
       if (exp - 300 > now) {
         this.log?.debug(`getAccessToken: ${source} => Token valid and simply returned`);
-        return this.accessToken;
+        if (!this.config.cookieAuth) return this.accessToken;
+        else return;
```

Comment:
I'd recommend some braces here, or a ternary. We should explicitly specify the return value too.